src/formatDate.js

```diff
@@ -50,7 +50,51 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const firstSeperator = fromFormat.pop();
+  const secondSeperator = toFormat.pop();
+
+  const separatedDate = date.split(`${firstSeperator}`);
+
+  const values = {};
+
+  separatedDate.forEach((el, index) => {
+    values[fromFormat[index]] = el;
+  });
+
+  if (values.hasOwnProperty('YYYY')) {
+    values.YY = values.YYYY.slice(2);
+  } else {
+    values.YYYY = (+values.YY < 30 && values.YY >= 0)
+      ? `20${values.YY}`
+      : `19${values.YY}`;
+  }
+
+  const formated = [];
+
+  toFormat.forEach(el => {
+    switch (el) {
+      case 'DD':
+        formated.push(values.DD);
+        break;
+
+      case 'MM':
+        formated.push(values.MM);
+        break;
+
+      case 'YY':
+        formated.push(values.YY);
+        break;
+
+      case 'YYYY':
+        formated.push(values.YYYY);
+        break;
+    }
+  });
+
+  fromFormat.push(firstSeperator);
+  toFormat.push(secondSeperator);
```

Comment:
nice, but you may do it much simpler, just make
`const firstSeperator = fromFormat[fromFormat.length - 1]`
or 
`const firstSeperator = fromFormat.slice().pop()`